### PR TITLE
Date filters on the analytics page

### PR DESCRIPTION
**Pull request by Becky-Hu-82**
### Summary 

This pull request implements the date filters on the analytics page.

- implemented a month picker and a year picker
- implemented a date range picker
- added the analytics overview component and its functionality
image
image

### CODE REVIEW

**Review by mg2239 — CHANGES_REQUESTED** [review]
can you better align these to the figma styles?

**Comment by Becky-Hu-82** [comment]
> I was wondering if you can use the icons (i.e. the calendar icon) provided in the Figma to customize the date-picker? (I can't run this on my end at the moment because of some error). But other than that it looks great to me!

That's a good point! Just changed the icon!

**Review by mg2239 — CHANGES_REQUESTED** [review]
image
im getting a dependency error with react-dropdown-date and our version of react. can you also try reinstalling dependencies and see? if its also happening on your end, would it be possible to use native html inputs instead?

File: frontend/src/components/AnalyticsTable/AnalyticsTable.tsx [hunk 1]
```diff
@@ -73,6 +77,94 @@ const Table = ({ type }: TableProps) => {
...
+      // should never get here
+      return [{
+        icon: dayRide,
+        alt: 'day',
+        stats: 0,
+        description: 'day rides',
+      }]
+    }
+  }
+
```
Comment (mg2239) [inline frontend/src/components/AnalyticsTable/AnalyticsTable.tsx:167]: this logic should be in the analytics overview component, not in the table component

File: frontend/src/components/AnalyticsTable/AnalyticsTable.tsx [hunk 2]
```diff
@@ -208,6 +300,9 @@ const Table = ({ type }: TableProps) => {
   }, [type, drivers.length]);
 
   return (
+    <>
+    <DateFilter/>
+    <AnalyticsOverview stats={generateStatsArr()} />
```
Comment (mg2239) [inline frontend/src/components/AnalyticsTable/AnalyticsTable.tsx:305]: move these to the analytics page in Admin. when connecting to backend, the analytics page will handle getting stats and passing the table data to all components, including the table and analytics overview

**Review by mg2239 — CHANGES_REQUESTED** [review]
overall looks good, but just a few changes to better separate the component logic

**Review by mg2239 — APPROVED** [review]
swapped out inputs to just be native, other changes require a lot of refactoring so that can be handled in a separate PR

**Pull request merged**